frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -0,0 +1,363 @@
+				"requestid":     "frontend.lint",
+				"build-arg:BAR": "678",
+			},
+			Frontend: "dockerfile.v0",
+		})
+		require.NoError(t, err)
+
+		lintResults, err := unmarshalLintResults(res)
+		require.NoError(t, err)
+
+		require.Equal(t, len(expected), len(lintResults.Warnings))
+		sort.Slice(lintResults.Warnings, func(i, j int) bool {
+			// sort by line number in ascending order
+			firstRange := lintResults.Warnings[i].Location[0]
+			secondRange := lintResults.Warnings[j].Location[0]
+			return firstRange.Start.Line < secondRange.Start.Line
+		})
+		// Compare expectedLintWarning with actual lint results
+		for i, w := range lintResults.Warnings {
+			checkLintWarning(t, w, expected[i])
+		}
+		called = true
+		return nil, nil
+	}
+
+	_, err = c.Build(sb.Context(), client.SolveOpt{
+		LocalMounts: map[string]fsutil.FS{
+			dockerui.DefaultLocalNameDockerfile: dir,
+		},
+	}, "", frontend, nil)
+	require.NoError(t, err)
+
+	require.True(t, called)
+}
+
+func checkProgressStream(t *testing.T, sb integration.Sandbox, c *client.Client, f frontend, dir *integration.TmpDirWithName, dockerfile []byte, expected []expectedLintWarning) {
+	status := make(chan *client.SolveStatus)
+	statusDone := make(chan struct{})
+	done := make(chan struct{})
+
+	var warnings []*client.VertexWarning
+
+	go func() {
+		defer close(statusDone)
+		for {
+			select {
+			case st, ok := <-status:
+				if !ok {
+					return
+				}
+				warnings = append(warnings, st.Warnings...)
+			case <-done:
+				return
+			}
+		}
+	}()
+
+	_, err := f.Solve(sb.Context(), c, client.SolveOpt{
+		FrontendAttrs: map[string]string{
+			"platform": "linux/amd64,linux/arm64",
+		},
+		LocalMounts: map[string]fsutil.FS{
+			dockerui.DefaultLocalNameDockerfile: dir,
+			dockerui.DefaultLocalNameContext:    dir,
+		},
+	}, status)
+	require.NoError(t, err)
+
+	select {
+	case <-statusDone:
+	case <-time.After(10 * time.Second):
+		t.Fatalf("timed out waiting for statusDone")
+	}
+
+	// two platforms only show one warning
+	require.Equal(t, len(expected), len(warnings))
+	sort.Slice(warnings, func(i, j int) bool {
+		w1 := warnings[i]
+		w2 := warnings[j]
+		if len(w1.Range) == 0 {
+			return true
+		} else if len(w2.Range) == 0 {
+			return false
+		}
+		return w1.Range[0].Start.Line < w2.Range[0].Start.Line
+	})
+	for i, w := range warnings {
+		checkVertexWarning(t, w, expected[i])
+	}
+}
+
+func checkLinterWarnings(t *testing.T, sb integration.Sandbox, dockerfile []byte, expected []expectedLintWarning) {
+	// As a note, this test depends on the `expected` lint
+	// warnings to be in order of appearance in the Dockerfile.
+
+	integration.SkipOnPlatform(t, "windows")
+	f := getFrontend(t, sb)
+
+	dir := integration.Tmpdir(
+		t,
+		fstest.CreateFile("Dockerfile", dockerfile, 0600),
+	)
+
+	c, err := client.New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	checkProgressStream(t, sb, c, f, dir, dockerfile, expected)
```

Comment:
wrap these with

```
t.Run("warntype=progress", func(t *testing.T){})
```

so they show up as separate subtests.

Looks like you could also call it via `range` over a `map[string]func(testing.T, ...)`.